tests/infra_test.go

```diff
@@ -95,6 +99,102 @@ var _ = Describe("[Serial][sig-compute]Infrastructure", func() {
 		}
 	})
 
+	Describe("changes to the kubernetes client", func() {
+
+		BeforeEach(func() {
+			tests.BeforeTestCleanup()
+		})
+
+		scheduledToRunning := func(vmis []v1.VirtualMachineInstance) time.Duration {
+			var duration time.Duration
+			for _, vmi := range vmis {
+				start := metav1.Time{}
+				stop := metav1.Time{}
+				for _, timestamp := range vmi.Status.PhaseTransitionTimestamps {
+					if timestamp.Phase == v1.Scheduled {
+						start = timestamp.PhaseTransitionTimestamp
+					} else if timestamp.Phase == v1.Running {
+						stop = timestamp.PhaseTransitionTimestamp
+					}
+				}
+				duration += stop.Sub(start.Time)
+			}
+			return duration
+		}
+
+		It("on the controller rate limiter should lead to delayed VMI starts", func() {
+			By("first getting the basetime for a replicaset")
+			replicaset := tests.NewRandomReplicaSetFromVMI(libvmi.NewCirros(libvmi.WithResourceMemory("1Mi")), int32(0))
+			replicaset, err = virtClient.ReplicaSet(util.NamespaceTestDefault).Create(replicaset)
+			Expect(err).ToNot(HaveOccurred())
+			start := time.Now()
+			libreplicaset.DoScaleWithScaleSubresource(virtClient, replicaset.Name, 10)
+			fastDuration := time.Now().Sub(start)
+			libreplicaset.DoScaleWithScaleSubresource(virtClient, replicaset.Name, 0)
+
+			By("reducing the throughput on controller")
+			originalKubeVirt := util.GetCurrentKv(virtClient)
+			originalKubeVirt.Spec.Configuration.ControllerConfiguration = &v1.ReloadableComponentConfiguration{
+				RestClient: &v1.RESTClientConfiguration{
+					RateLimiter: &v1.RateLimiter{
+						TokenBucketRateLimiter: &v1.TokenBucketRateLimiter{
+							Burst: 3,
+							QPS:   2,
+						},
+					},
+				},
+			}
+			tests.UpdateKubeVirtConfigValueAndWait(originalKubeVirt.Spec.Configuration)
+			By("starting a replicaset with reduced throughput")
+			start = time.Now()
+			libreplicaset.DoScaleWithScaleSubresource(virtClient, replicaset.Name, 10)
+			slowDuration := time.Now().Sub(start)
+			Expect(slowDuration.Seconds()).To(BeNumerically(">", 5*fastDuration.Seconds()))
+		})
+
+		It("on the virt handler rate limiter should lead to delayed VMI running states", func() {
```

Comment:
This kind of test I was thinking we should leave to be measured by the perf/scale framework.  QPS improves performance by reducing the delta between when a virt-launcher pod is Running and when a VMI is Running, which is perfect for a threshold test using the new framework. 

For a func test to validate this feature, I was thinking a test that checks the number of times we get rate limited with low and high qps.  That will tell us if our default values ever get too low again or if they are too high because we made a fix that lowers the number of API calls we make.

Reply:
> This kind of test I was thinking we should leave to be measured by the perf/scale framework. QPS improves performance by reducing the delta between when a virt-launcher pod is Running and when a VMI is Running, which is perfect for a threshold test using the new framework.

Definitely. This test does not imply that that these values are good or bad and they are not subject of change. They should simply state that the rate limiter settings have an effect which makes things slower.

Reply:
Right, but the most accurate thing to measure here is the number of times we hit the rate limiter based on the QPS.  Slowness is a symptom that can be affected by many things, so I want to leave that measurement to the perf framework

Reply:
> Slowness is a symptom that can be affected by many things

Yes, that is why this takes a baseline first.

I understand what you want to get at, still I want to ask for doing this with reasonable complexity. It is just pretty simple and should be pretty robust this way, while going to the metric and trying to interpret it is more complex.

Reply:
Since the measured latencies are relative, the test does not characterize performance.

Although the @rthallisey idea of counting the rate limited calls is very interesting, it requires collect prometheus metrics which complicates the test. First, prometheus is not deployed by default in the functional tests. Second, it requires more coding...